airbyte-cdk/python/airbyte_cdk/sources/connector_state_manager.py

```diff
@@ -0,0 +1,44 @@
+#
+# Copyright (c) 2022 Airbyte, Inc., all rights reserved.
+#
+
+from typing import Any, List, Mapping, MutableMapping, Union
+
+from airbyte_cdk.models import AirbyteStateBlob, AirbyteStateMessage, AirbyteStateType
+
+
+class ConnectorStateManager:
+    """
+    ConnectorStateManager consolidates the various forms of a stream's incoming state message (STREAM / GLOBAL / LEGACY) under a common
+    interface. It also provides methods to extract and update state
+    """
+
+    # In the immediate, we only persist legacy which will be used during abstract_source.read(). In the subsequent PRs we will
+    # initialize the ConnectorStateManager according to the new per-stream interface received from the platform
```

Comment:
I think an equivalent comment should be made on `AbstractSource.read`, wdyt?

Reply:
actually i think it's okay to just comment here since this is meant to provide a little why state manager is slimmed down. wheras in abstract_source.py, the code is sort of self explanatory since we didn't have to add any extra boilerplate. We'll clear this stuff out in a few days as well